acceptance-tests/src/main/kotlin/org/ostelco/at/okhttp/Tests.kt

```diff
@@ -994,7 +996,125 @@ class PlanTest {
             purchaseRecords.sortBy { it.timestamp }
 
             assert(Instant.now().toEpochMilli() - purchaseRecords.last().timestamp < 10_000) { "Missing Purchase Record" }
-            assertEquals(expectedPlanProduct, purchaseRecords.last().product, "Incorrect 'Product' in purchase record")
+            assertEquals(expectedPlanProductSG, purchaseRecords.last().product, "Incorrect 'Product' in purchase record")
+        } finally {
+            StripePayment.deleteCustomer(customerId = customerId)
+        }
+    }
+}
+
+class RenewPlanTest {
+
+    @Test
+    fun `okhttp test - POST purchase plan with trial time`() {
+
+        val email = "purchase-${randomInt()}@test.com"
+        val sku = "PLAN_10USD_DAY"
+        var customerId = ""
+
+        try {
+            customerId = createCustomer(name = "Test Purchase Plan User", email = email).id
+            enableRegion(email = email, region = "us")
+
+            val client = clientForSubject(subject = email)
+            val sourceId = StripePayment.createPaymentTokenId()
+
+            client.purchaseProduct(sku, sourceId, false)
+
+            // Wait for DB to be updated.
+            Thread.sleep(200)
+
+            var purchaseRecords = client.purchaseHistory
+            purchaseRecords.sortBy { it.timestamp }
+
+            assert(Instant.now().toEpochMilli() - purchaseRecords.last().timestamp < 10_000) { "Missing purchase record" }
+
+            // First record - free product
+            // Second record - product subscribed to
+            assert(2 == purchaseRecords.size) { "Got ${purchaseRecords.size} purchase records, expected 2" }
+            assertEquals(expectedPlanProductUS, purchaseRecords.last().product, "Incorrect 'Product' in purchase record")
+
+            // Actual charge for renewal will first be done after trial time
+            // expires, which will happen after 4 sec. Waiting a bit longer
+            // before checking the outcome.
+            StripeEventListener.waitForSubscriptionPaymentToSucceed(
+                    subscription = "stripe-event-okhttp-purchase-ok-sub",
+                    customerId = customerId,
+                    timeout = 30000L)
+
+            // Wait for DB to be updated.
+            Thread.sleep(200)
```

Comment:
Sleeping is a very brittle way to ensure that a database is updated. Is there a better way?